strawberry/tools/create_type.py

```diff
@@ -33,14 +34,15 @@ def create_type(name: str, fields: List[StrawberryField]) -> Type:
                 )
             )
 
-        dataclass_fields.append(
-            (
-                field.graphql_name,
-                field.type,
-                dataclasses_field(default=field),
-            )
-        )
+    for field in fields:
+        if not isinstance(field, StrawberryField):
+            raise TypeError("Field is not an instance of StrawberryField")
```

Comment:
Do we want to include the field name/qualname in the error message?

Reply:
maybe! this code is duplicated by the way, bad rebase :D